pandas/core/reshape/tile.py

```diff
@@ -498,13 +503,15 @@ def _convert_bin_to_datelike_type(bins, dtype):
     return bins
 
 
-def _format_labels(bins, precision, right=True, include_lowest=False, dtype=None):
+def _format_labels(
+    bins, precision, right: bool = True, include_lowest: bool = False, dtype=None
+):
     """ based on the dtype, return our labels """
 
     closed = "right" if right else "left"
 
     if is_datetime64tz_dtype(dtype):
-        formatter = partial(Timestamp, tz=dtype.tz)
+        formatter = lambda x: Timestamp(x, tz=dtype.tz)
```

Comment:
What error was this throwing? I would think a partial to be preferable to a lambda

Reply:
`error: Incompatible types in assignment (expression has type "Callable[[Any], Any]", variable has type "partial[Any]")`

Reply:
Hmm that's unfortunate. I suppose goes back to https://github.com/python/mypy/issues/1484